apheleia-formatters.el

```diff
@@ -2,10 +2,102 @@
 
 ;;; Commentary:
 
-;; Formatter definitions for `apheleia'.
+;; Formatter helper functions and definitions for `apheleia'.
 
 ;;; Code:
 
+(require 'cl-lib)
+
+(defun apheleia-formatters-indent (tab-flag indent-flag indent-var)
+  "Set flag for indentation.
+Helper function for `apheleia-formatters' which allows you to supply
+alternating flags based on the current buffers indent configuration. If the
+buffer is indented with tabs then returns TAB-FLAG. Otherwise if INDENT-VAR
+is set in the buffer return INDENT-FLAG and the value of INDENT-VAR. Use this
+to easily configure the indentation level of a formatter."
+  (cond
+   (indent-tabs-mode tab-flag)
+   (indent-var
+    (when-let ((indent (and (boundp indent-var)
+                            (symbol-value indent-var))))
+      (list indent-flag (number-to-string indent))))))
+
+(defun apheleia-formatters-js-indent (tab-flag indent-flag)
+  "Variant of `apheleia-formatters-indent' for JavaScript like modes.
+See `apheleia-formatters-indent' for a description of TAB-FLAG and
+INDENT-FLAG."
+  (apheleia-formatters-indent
+   tab-flag indent-flag
+   (cl-case major-mode
+     (json-mode 'js-indent-level)
+     (json-ts-mode 'json-ts-mode-indent-offset)
+     (js-mode 'js-indent-level)
+     (js-jsx-mode 'js-indent-level)
+     (js2-mode 'js2-basic-offset)
+     (js2-jsx-mode 'js2-basic-offset)
+     (js3-mode 'js3-indent-level))))
```

Comment:
This is kind of an annoying maintenance burden. I don't want to get to the point where we're maintaining the associations between various modes and indent variables :/. I normally use [editorconfig](https://github.com/editorconfig/editorconfig-emacs/blob/e1a391a618ec33d157822dbcc51d010559289f1a/editorconfig.el#L180) for this. If you're okay with adding a dependency on it then I think we'd get much more comprehensive coverage in apheleia through it.

Reply:
I think it's perfectly fine to add a dependency on editorconfig if this allows indentation settings to be respected by formatters. Thanks for the idea.

Reply:
I'll defer this to a follow up.